server/apps/archive/archive.py

```diff
@@ -362,17 +367,15 @@ def _duplicate_item(self, original_doc):
         del new_doc[config.ID_FIELD]
         del new_doc['guid']
 
+        new_doc[ITEM_OPERATION] = ITEM_DUPLICATE
         item_model = get_model(ItemModel)
 
         on_duplicate_item(new_doc)
-        item_model.create([new_doc])
-        self._duplicate_versions(original_doc['guid'], new_doc)
-
+        resolve_document_version(new_doc, 'archive', 'PATCH', new_doc)
```

Comment:
1. Yes, you were right. the `_duplicate_versions()` inserts the last version.
2. Not sure if POs agree to increment the version for recording the operation.
3. I looked into `duplicate_item()` it does increment the version irrespective of state. However, as mentioned by you we need extra version for recording the operation.